core/dropfile.js

```diff
@@ -68,185 +82,141 @@ module.exports = class DropFile {
     }
 
     getHandler() {
-        return {
-            DOOR: this.getDoorSysBuffer,
-            DOOR32: this.getDoor32Buffer,
-            DORINFO: this.getDoorInfoDefBuffer,
-        }[this.fileType];
+        // TODO: Replace with a switch statement once we have binary handlers as well
+
+        // Read the directory containing the dropfile formats, and return undefined if we don't have the format
+        const fileName = this.fileName;
+        if (!fileName) {
+            Log.info({fileType: this.fileType}, 'Dropfile format not supported.');
+            return undefined;
+        }
+        const filePath = paths.join(this.dropFileFormatDirectory, fileName);
+        if(!fs.existsSync(filePath)) {
+            Log.info({filename: fileName}, 'Dropfile format not found or readable.');
+            return undefined;
+        }
+
+        // Return the handler to get the dropfile, because in the future we may have additional handlers
+        return this.getDropfile;
     }
 
     getContents() {
-        const handler = this.getHandler().bind(this);
-        return handler();
+        const handlerRef = this.getHandler();
+        if(!handlerRef) {
+            return undefined;
+        }
+        const handler = handlerRef.bind(this);
+        const contents = handler();
+        return contents;
     }
 
-    getDoorInfoFileName() {
-        let x;
-        const node = this.client.node;
-        if (10 === node) {
-            x = 0;
-        } else if (node < 10) {
-            x = node;
-        } else {
-            x = String.fromCharCode('a'.charCodeAt(0) + (node - 11));
+    getDropfile() {
+        // Get the filename to read
+        const fileName = paths.join(this.dropFileFormatDirectory, this.fileName);
+
+        let text = fs.readFileSync(fileName);
+
+        // Format the data with string_format and predefined_mci
+        let formatObj = getPredefinedMCIFormatObject(this.client, text);
+
+        const additionalFormatObj = {
+            'getSysopFirstName': this.getSysopFirstName(),
+            'getSysopLastName': this.getSysopLastName(),
+            'getUserFirstName': this.getUserFirstName(),
+            'getUserLastName': this.getUserLastName(),
+            'getUserTotalDownloadK': this.getUserTotalDownloadK(),
+            'getUserTotalUploadK': this.getUserTotalUploadK(),
+            'getCurrentDateMMDDYY': this.getCurrentDateMMDDYY(),
```

Comment:
We might be able to do it here as well, something like `{currentDate!dateMmDdYy}` or something, with some other common formats.